src/main/java/gregtech/api/capability/impl/HeatingCoilRecipeLogic.java

```diff
@@ -0,0 +1,47 @@
+package gregtech.api.capability.impl;
+
+import gregtech.api.capability.IHeatingCoil;
+import gregtech.api.metatileentity.multiblock.RecipeMapMultiblockController;
+import gregtech.api.recipes.Recipe;
+import gregtech.api.recipes.recipeproperties.BlastTemperatureProperty;
+
+import javax.annotation.Nonnull;
+
+/**
+ * RecipeLogic for multiblocks that use temperature for raising speed and lowering energy usage
+ * Used with RecipeMaps that run recipes using the {@link gregtech.api.recipes.recipeproperties.BlastTemperatureProperty}
+ */
+public class HeatingCoilRecipeLogic extends MultiblockRecipeLogic {
+
+    public HeatingCoilRecipeLogic(RecipeMapMultiblockController metaTileEntity) {
+        super(metaTileEntity);
+    }
+
+    @Override
+    protected int[] runOverclockingLogic(@Nonnull Recipe recipe, boolean negativeEU, int maxOverclocks) {
+        return blastFurnaceOverclockingLogic(recipe.getEUt(), getMaxVoltage(), recipe.getDuration(), maxOverclocks,
+                ((IHeatingCoil) metaTileEntity).getCurrentTemperature(),
+                recipe.getProperty(BlastTemperatureProperty.getInstance(), 0));
+    }
+
+    @Nonnull
+    public static int[] blastFurnaceOverclockingLogic(int recipeEUt, long maximumVoltage, int recipeDuration, int maxOverclocks, int currentTemp, int recipeRequiredTemp) {
```

Comment:
If you are trying to change this into a class that could be used for other multiblocks, this method could use a rename so that it is not EBF specific. Maybe something like `heatOverclockingLogic`